lib/ansible/module_utils/ec2.py

```diff
@@ -460,12 +459,17 @@ def boto3_tag_list_to_ansible_dict(tags_list, tag_name_key_name='Key', tag_value
         }
     """
 
-    tags_dict = {}
-    for tag in tags_list:
-        if tag_name_key_name in tag:
-            tags_dict[tag[tag_name_key_name]] = tag[tag_value_key_name]
-
-    return tags_dict
+    if tag_name_key_name and tag_value_key_name:
```

Comment:
I think we could just ignore these parameters entirely, and assume we'll need to try both all the time. It would simplify this code, and let us list those params as deprecated.

Reply:
I'm not sure I understand this.  When you say 'we'll need to try both all the time', both what?

I think we need these parameters because there are multiple variations that AWS use for tag key names.

Reply:
I've just checked the source, and we don't actually pass values to boto3_tag_list_to_ansible_dict anywhere at present!

Is this a practical concern or a theoretical concern - are there any services that don't use `[{'Key': tag_key, 'Value': tag_value}]` ?

I know that using the way the return result of `compare_aws_tags` is used does greatly depend on the service in question, but really don't know if there are similar concerns here.

Reply:
The way I'm thinking about it, we have the choice between:

A: trying and falling back on a known alternative (lower-case `key`), thus *not* adding parameters to this function and avoiding more surface area

B: adding these params to handle a case we don't have an example of

To me, option A seems like the way to go if we're able to get away with it, and we can just remove the usage of these kwargs in those instance/snapshot/etc modules. Does that make sense to y'all?

Reply:
Ah, mild issue: since this is also going to be a backport, we shouldn't remove the kwarg usage entirely, just ignore the parameters for same effect.

Reply:
This is weird. I definitely remember coding a module that didn't use Key or key. I thought it was merged but maybe not. 

We definitely need to keep the parameters to specify. Previously this module did what is proposed as (a) but I changed it because it doesn't satisfy all cases

Reply:
Any objections to just merging this as is then?

Reply:
LGTM